src/riakc_obj.erl

```diff
@@ -494,6 +498,26 @@ add_secondary_index(MD, [{Id, BinList} | Rest]) when is_binary(Id) ->
             add_secondary_index(MD2, Rest)
     end.
 
+%% @doc  Clear TTL entry
+-spec clear_ttl(metadata()) -> metadata().
+clear_ttl(MD) ->
+    dict:erase(?MD_TTL, MD).
+
+%% @doc  Get TTL
+-spec get_ttl(metadata()) -> ttl() | undefined.
+get_ttl(MD) ->
+    case dict:find(?MD_TTL, MD) of
+        {ok, TTL} ->
+            TTL;
+        error ->
+            false
```

Comment:
Huh, very weird. I wonder why dialyzer didn't catch that. Seems really obvious, and I don't see any ignore-warnings files either. I'll fix the spec but I'm also going to try and spend a little time figuring out why dialyzer is not flagging that code....